### PR TITLE
chore(tests): include junit,verbose reporters and junit report output…

### PR DESCRIPTION
**Pull request by odockal**
… file

### What does this PR do?
Adds new reporter (junit) and verbose one, which should mostly be valuable when there are tests failures. It also produces a junit report xml file under `tests/output`, it does not introduce any new location or anything. It can be processed on CI then.

### What issues does this PR fix or reference?
#3292 

### How to test this PR?
`yarn test:e2e:smoke` -> there will be `tests/output/junit-results.xml` file with all test cases results.

### CODE REVIEW

**Review by benoitf — CHANGES_REQUESTED** [review]
the default reporter is much more easier for development as it collapse suites when they pass

I guess here it's a matter of personal preference 

I suspect we need to find a way to allow people choose their reporter without changing for all person

`yarn run test:e2e:smoke --reporter junit` is not working ?

**Comment by odockal** [comment]
> the default reporter is much more easier for development as it collapse suites when they pass
> I guess here it's a matter of personal preference

Totally agree. I would just like to emphasize that we are talking only about e2e tests here.

Cli should work as well.

If we change reporters to 'default' and 'junit' with an outputfile definition for junit reporter, log output will stay the same and just junit file will be generated. 

so this one could be ok:
```
    reporters: ['default', 'junit'],
    outputFile: { junit: 'tests/output/junit-results.xml' },
```

Or this with a CI env. var taken into account:
```
    reporters: ['default', process.env.CI ? 'junit' : ''],
    outputFile: process.env.CI ? { junit: 'tests/output/junit-results.xml' } : {},
```
I would prefer this options before using cli as the string in npm task is growing. WDYT?

**Comment by benoitf** [comment]
if you need these results only for CI, yes the later option seems to be the better

File: vitest.config.js [hunk 1]
```diff
@@ -47,6 +47,9 @@ const config = {
      */
     testTimeout: 60_000,
     hookTimeout: 60_000,
+    // test reporters - default for all and junit for CI
+    reporters: ['default', process.env.CI ? 'junit' : ''],
```
Comment (benoitf) [inline vitest.config.js]: does it work with an empty reporter ?

without ci it will do ['default', '']

Comment (odockal) [reply vitest.config.js]: Seems yes!
Screenshot_20230808_165906

But it does not look good, I agree. Changed to readable form without empty reporter.

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
thanks

**Comment by odockal** [comment]
@benoitf Hm, I was checking the file available on GHA, it contains totally unrelated test results. so I am back to investigate what is going on.

**Comment by odockal** [comment]
@benoitf I have noticed two more issues with reporter and e2e tests config, first: `tools` folder was included in vitest.config and so I updated included files and second, there was same name for artifact saved for whole workflow file. Overlapping names. Will check results.

**Comment by odockal** [comment]
@benoitf E2e test artifact archive now contains right junit.xml file. Contains just e2e test results.

**Comment by benoitf** [comment]
There is now a unit-test artifact with 0 bytes so somehow either we need to publish something or we don't create the artifact ?

Screenshot 2023-08-10 at 09 41 16

**Comment by odockal** [comment]
@benoitf true, removing unit tests archive artifacts step. For now I do not see a point why to store result from unit tests, wdyt?

**Review by benoitf — APPROVED** [review]

**Pull request merged**